velox/exec/MergingVectorInput.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/type/Type.h"
+#include "velox/vector/ComplexVector.h"
+
+namespace facebook::velox::exec {
+
+class MergingVectorInput {
+ public:
+  MergingVectorInput(
+      velox::memory::MemoryPool* pool,
+      int64_t preferredOutputBatchBytes,
+      int32_t preferredOutputRows,
+      int32_t maxMergingRows,
+      bool preserveOrder);
+
+  /// Add input vector. It will be merged into a big vector if smaller than
```

Comment:
We may need to have 2 limits: number of rows and number of bytes. When processing wide data (lots of columns or columns are of complex types with lots of nested values) a small number of rows may use a lot of bytes.

Reply:
Done.